### PR TITLE
Scripts/IcecrownCitadel: Adding IsAlive check to avoid crash in Event

### LINKED ISSUE #24505: [ICC] Crash Boss_Svalna (captains) | Motionmaster
Server crashes all of a sudden in ICC on boss_sister_svalna. Related to Motionmaster (maybe?) crash logs always includes names:

npc_captain_arnath
npc_captain_brandon
npc_captain_grondel
npc_captain_rupert

**Branch(es):**
3.3.5

**TC rev. hash/commit:** 
b4584ee32380b6a01c681f86e59a8b8258ba39fb

*** Hardware ***
Processor: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
Number Of Processors: 8
Physical Memory: 33187668 KB (Available: 25510008 KB)
Commit Charge Limit: 37381972 KB

*** Operation System ***
Windows Server 2012 Standard (Version 6.2, Build 9200)

Crash Log:
https://drive.google.com/open?id=1oNpemu0kwnHM1PL3ZiwlrK0k-DOTrqKQ

### PR DESCRIPTION
**Pull request by Faq**
thx to participants in #24505 for solving this

**Changes proposed:**

-  Adding IsAlive check to avoid crash in Event

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #  (insert issue tracker number)
closes #24505

**Tests performed:** (Does it build, tested in-game, etc.)
compiles, tests performed

### CODE REVIEW

**Comment by Jildor** [comment]
@Faq Have you been able to reproduce the freeze? I can't reproduce it as show in the video: https://github.com/TrinityCore/TrinityCore/issues/24505#issuecomment-620962207

**Comment by Shauren** [comment]
*crashes*, not freezes, that guy cant even tell the difference

**Comment by jackpoz** [comment]
@Faq please include a crashlog using a clean 3.3.5 as the issue didn't have any

File: src/server/scripts/Northrend/IcecrownCitadel/boss_sister_svalna.cpp [hunk 1]
```diff
@@ -568,7 +568,8 @@ struct npc_crok_scourgebane : public EscortAI
 
             for (uint32 itr = 0; itr < 4; ++itr)
                 if (Creature* crusader = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_CAPTAIN_ARNATH + itr)))
-                    crusader->AI()->DoAction(ACTION_START_GAUNTLET);
+                    if (crusader->IsAlive())
+                        crusader->AI()->DoAction(ACTION_START_GAUNTLET);
```
Comment (jackpoz) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_sister_svalna.cpp:572]: if you check the lines above at https://github.com/TrinityCore/TrinityCore/pull/24514/files#diff-834d6b45f7457540ff0fb85af4e413d2R548-R550 , the IsAlive() check is done for this script in the called DoAction() handler, not in the caller. Would it make sense to do the same for Crusader adding a IsAlive() check there ?

Comment (Faq) [reply src/server/scripts/Northrend/IcecrownCitadel/boss_sister_svalna.cpp:572]: true

**Comment by Faq** [comment]
Cant reproduce, note to myself, don't trust so easily. No doubt there are custom things n their core or not all commits pulled.

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Comment by ckegg** [comment]
Can't reproduce either on commit 411033ddf819db9e8ac5c1e6c540cf529138b5e6 right before 22aaf32a79cd6ce8ebb4507cdfd677f073c35178 , not sure why it froze a while back. So hopefully it is all good now

**Comment by jackpoz** [comment]
we couldn't reproduce it either, the IsAlive() check made sense anyway 😃

**Comment by ckegg** [comment]
I made it. Was able to reproduce the freezes on commit b48c50ba23fc9c12b8e9d6a7d0c5d65ffd4b9a59
[crash.log](https://github.com/TrinityCore/TrinityCore/files/4567118/crash.log)

**Comment by Faq** [comment]
Any video to how reproduce? But please post on https://github.com/TrinityCore/TrinityCore/issues/24505

**Comment by jackpoz** [comment]
well, that's a crash before this PR so it's fine now :)

**Pull request merged**